abn/metricstore/metric.go

```diff
@@ -0,0 +1,258 @@
+package metricstore
+
+import (
+	"context"
+	"errors"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/iter8-tools/iter8/base/log"
+	"github.com/iter8-tools/iter8/driver"
+	"sigs.k8s.io/yaml"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+const (
+	VERSIONS_DATA = "versionData.yaml"
+)
+
+type ApplicationData map[string]VersionData
+
+type VersionData struct {
+	Metrics             map[string]SummaryMetric `json:"metrics,omitempty" yaml:"metrics,omitempty"`
+	LastUpdateTimestamp time.Time                `json:"lastUpdateTimestamp" yaml:"lastUpdateTimestamp"`
+	History             []VersionEvent           `json:"history,omitempty" yaml:"history,omitempty"`
+}
+
+type VersionEvent struct {
```

Comment:
I don't think we need to worry about Kubernetes objects that are improperly labeled/annotated. Iter8 should clearly document the labels/annotations it expects -- and if they are not present, the kubernetes object should be considered to be outside the purview of abn experimentation / service -- to be simply ignored by Iter8!
***
We are asking for a few labels, track, version, app, ready or not. This is really reasonable for a feature that deals with AB testing, and handles both metrics and routing.
***
We will use this approach for autox also -- if you supply the labels we require, some work will happen. Otherwise an expensive no-op will happen!

Reply:
Is this comment associated with the referenced code in some way? I do not see the relationship of the comment to the code. in general, this was the approach I took though as you do comment in a couple of other places, I did try to be a little flexible. Some of this was meant to be removed so ... thanks for catching.

Reply:
I saw code that tried to assign version identifiers to tracks (in case the latter was missing). This is not desirable, since versions and tracks are distinct concepts. The right thing to do, IMO, if Kubernetes objects are not annotated properly, is to simply leave them alone (we can log ... but then we need a command to surface the abn logs easily to the user).